docs/source/trainer/time.rst

```diff
@@ -0,0 +1,106 @@
+Time
+====
+
+We use a |Time| class to represent and track time throughout
+the training loop. We track several quantities (epochs, batches,
+samples, and tokens) throughout training. Values
+can be provided as a string:
+
+- epochs: ``"10ep"``
```

Comment:
Adding on to #585, can we perhaps present this in a table form, where we list for each quantity:

- Its name
- Its enum value
- A b
- Its string form code (e.g. "ep", "ba", ...)
- An example of the string form

I think it would also be helpful to add a short description for each of Timestamp, Time, and TimeUnit (we already have the description for the state.Timer below)

Reply:
I think the table probably belongs in the docstrings. As a user guide, a table might be just a bit too complicated to grok -- users don't actually need to know the enum value, and the usage is pretty straightforward from the examples there.

Reply:
On second thought, made a table!